gitindex/index.go

```diff
@@ -532,6 +518,225 @@ func newIgnoreMatcher(tree *object.Tree) (*ignore.Matcher, error) {
 	return ignore.ParseIgnoreFile(strings.NewReader(content))
 }
 
+func prepareDeltaBuild(options Options, repository *git.Repository) (repos map[fileKey]BlobLocation, branchMap map[fileKey][]string, branchVersions map[string]map[string]plumbing.Hash, changedOrDeletedPaths []string, err error) {
+	// discover what commits we indexed during our last build
+	shards := options.BuildOptions.FindAllShards()
+	if len(shards) == 0 {
+		return nil, nil, nil, nil, fmt.Errorf("no existing shards found for repository")
+	}
+
+	shard := shards[0]
+	repositories, _, err := zoekt.ReadMetadataPathAlive(shard)
+	if err != nil {
+		return nil, nil, nil, nil, fmt.Errorf("reading reposistory metadata from shard %q: %w", shard, err)
+	}
+
+	var existingRepository *zoekt.Repository
+
+	for _, r := range repositories {
+		if r.ID == options.BuildOptions.RepositoryDescription.ID {
+			existingRepository = r
+			break
+		}
+	}
```

Comment:
we should assume the current shard is not compound. IE if we have multiple repos here, we return early to not use delta. I believe compound + delta would be very tricky. Possible in future, but better to keep it simple unless this works with just this simple change.

Reply:
So to clarify, we want to this to look like 

```
exitingRepository := repositories[0]
```

instead of using a for loop over the repositories list?